### PR TITLE
Add price bump config

### LINKED ISSUE #3945: Make fee bump rules configurable in txpool
### Describe the feature

currently, PRICE_BUMP is a constant:

https://github.com/paradigmxyz/reth/blob/f98b1524bba6ef3e48fbd421eb540ef363db8b27/crates/transaction-pool/src/pool/txpool.rs#L1105-L1109

we want a setting that allows us to configure this check:

https://github.com/paradigmxyz/reth/blob/f98b1524bba6ef3e48fbd421eb540ef363db8b27/crates/transaction-pool/src/pool/txpool.rs#L1018-L1021

two approaches
a) a config struct that has required fee bumps depending on the _existing_ txtype
b) a trait that provides this function which we can then box

however if we want b)'s implementation to also be configurable then we end up at a)

so I think we should go with a), a struct that has

* default price_bump (PRICE_BUMP == 10%)
* replace blob tx price bump (100%)

should be added as new struct to

https://github.com/paradigmxyz/reth/blob/f98b1524bba6ef3e48fbd421eb540ef363db8b27/crates/transaction-pool/src/config.rs#L10-L12

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by altugbakan**
Fixes #3945 

This pull request adds a `PriceBumpConfig`struct for making fee bump rules configurable.

### CODE REVIEW

File: crates/transaction-pool/src/config.rs [hunk 1]
```diff
@@ -57,3 +63,36 @@ impl Default for SubPoolLimit {
...
+    fn default() -> Self {
+        Self {
+            default_price_bump: Default::default(),
+            replace_blob_tx_price_bump: PriceBump(REPLACE_BLOB_PRICE_BUMP),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct PriceBump(u128);
```
Comment (mattsse) [inline crates/transaction-pool/src/config.rs:86]: while I like having `default::default`,

I think this adds more complexity than it is useful,

I'm okay with just using u128 as fields ins PriceBumpConfig

File: crates/transaction-pool/src/pool/txpool.rs [hunk 2]
```diff
@@ -1105,7 +1105,7 @@ impl<T: PoolTransaction> AllTransactions<T> {
                 if Self::is_underpriced(
                     transaction.as_ref(),
                     entry.get().transaction.as_ref(),
-                    PRICE_BUMP,
+                    PriceBumpConfig::default().default_price_bump.into(),
```
Comment (mattsse) [inline crates/transaction-pool/src/pool/txpool.rs:1108]: we want this as a field in PoolConfig

**Review by mattsse — CHANGES_REQUESTED** [review]
nice,

almost there already

**Comment by altugbakan** [comment]
@mattsse was 

```rust
if Self::is_underpriced(
    transaction.as_ref(),
    entry.get().transaction.as_ref(),
    PoolConfig::default().price_bump,
) {
```
what you meant?

File: crates/transaction-pool/src/pool/txpool.rs (hunk 2 above)
Comment (altugbakan) [reply crates/transaction-pool/src/pool/txpool.rs]: I made some changes but I'm not sure if it is the intended solution.

File: bin/reth/src/args/txpool_args.rs [hunk 3]
```diff
@@ -33,6 +33,10 @@ pub struct TxPoolArgs {
     /// Max number of executable transaction slots guaranteed per account
     #[arg(long = "txpool.max_account_slots", help_heading = "TxPool", default_value_t = TXPOOL_MAX_ACCOUNT_SLOTS_PER_SENDER)]
     pub max_account_slots: usize,
+
+    /// Price bump (in %) for the transaction pool underpriced check.
+    #[arg(long = "txpool.price_bump", help_heading = "TxPool", default_value_t = DEFAULT_PRICE_BUMP)]
+    pub price_bump: u128,
```
Comment (mattsse) [inline bin/reth/src/args/txpool_args.rs:39]: nice!

**Review by mattsse — APPROVED** [review]
awesome!

this is great,
once we support blob tx we can easily integrate this as well

**Pull request merged**